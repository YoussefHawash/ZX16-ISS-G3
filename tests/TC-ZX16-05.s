.text
.org 0x0000
    j main

.org 0x0020
main:
    li x4, 0
    j skip_block

    li x4, 60
    li x4, 60
    li x4, 60

skip_block:
    li x2, 25           # simulate return address offset
    add x2, x2          # just preserve it
    j func

after_func:
    li x3, 55           # confirm return from func
    j done              # explicitly continue to done

func:
    li x4, 44           # confirm func ran
    jr x2               # return

done:
    li x7, 60
    ecall 10