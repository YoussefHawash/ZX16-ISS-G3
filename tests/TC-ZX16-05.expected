J 32
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
ADD x0, x0
LI x4, 0x00
J 8
LI x4, 0x3c
LI x4, 0x3c
LI x4, 0x3c
LI x2, 0x19
ADD x2, x2
J 6
LI x3, 0x37
J 6
LI x4, 0x2c
JR  x2
LI x7, 0x3c
ECALL 10    # ProgramExit
ADD x0, x0
ADD x0, x0
ADD x0, x0